app/controllers/admin/email_controller.rb

```diff
@@ -11,7 +11,13 @@ def test
     params.require(:email_address)
     begin
       Jobs::TestEmail.new.execute(to_address: params[:email_address])
-      render body: nil
+      if SiteSetting.disable_emails == "yes"
+        render json: { sentTestEmailMessage: I18n.t("admin.email.sent_test_disabled") }
+      elsif SiteSetting.disable_emails == "non-staff" && !User.find_by_email(params[:email_address])&.staff?
+        render json: { sentTestEmailMessage: I18n.t("admin.email.sent_test_disabled_for_non_staff") }
+      else
+        render json: { sentTestEmailMessage: I18n.t("admin.email.sent_test") }
```

Comment:
Our convention is to use snake_case instead of camelCase on the server side even for keys in the payload. We should add a spec for this as well :smile_cat: